*** dtk-speak.el~	Thu May 15 11:01:59 2003
--- dtk-speak.el	Thu Apr  3 22:54:50 2003
***************
*** 168,173 ****
--- 168,175 ----
  ;;{{{ macros
  
  (defmacro tts-with-punctuations (setting &rest body)
+   ;; Changed by Milan Zamazal <pdm@brailcom.org>, March 2003, to allow use of
+   ;; speechd-el.
    "Safely set punctuation mode for duration of body form."
    (`
     (progn
***************
*** 176,192 ****
         (unwind-protect
             (progn
               (unless (string= (, setting) save-punctuation-mode)
!                (process-send-string dtk-speaker-process
!                                     (format "tts_set_punctuations %s  \n "
!                                             (, setting)))
                 (setq dtk-punctuation-mode (, setting)))
               (,@ body)
               (dtk-force))
           (unless (string=  (, setting)  save-punctuation-mode)
             (setq dtk-punctuation-mode save-punctuation-mode)
!            (process-send-string dtk-speaker-process
!                                 (format "tts_set_punctuations %s  \n "
!                                         dtk-punctuation-mode ))
             (dtk-force)))))))
  
  ;;}}}
--- 178,190 ----
         (unwind-protect
             (progn
               (unless (string= (, setting) save-punctuation-mode)
!                (dtk-interp-set-punctuations (, setting))
                 (setq dtk-punctuation-mode (, setting)))
               (,@ body)
               (dtk-force))
           (unless (string=  (, setting)  save-punctuation-mode)
             (setq dtk-punctuation-mode save-punctuation-mode)
!            (dtk-interp-set-punctuations dtk-punctuation-mode)
             (dtk-force)))))))
  
  ;;}}}
***************
*** 786,788 ****
--- 784,787 ----
  ;;; end:
  
  ;;}}}
+ 
*** dtk-tcl.el~	Wed May 28 10:22:43 2003
--- dtk-tcl.el	Wed May 28 10:20:18 2003
***************
*** 473,479 ****
          (setq start  last
                personality (get-text-property last  'personality))) ; end while
        ))                                ; end clause
!    (t (dtk-interp-queue (buffer-substring start end  )))))
  
                                          ;Force the speech.
  (defsubst dtk-force ()
--- 473,479 ----
          (setq start  last
                personality (get-text-property last  'personality))) ; end while
        ))                                ; end clause
!    (t (dtk-interp-queue (buffer-substring-no-properties start end  )))))
  
                                          ;Force the speech.
  (defsubst dtk-force ()
***************
*** 551,556 ****
--- 551,559 ----
    "*Scale factor applied to speech rate when skimming.")
  
  (defun dtk-speak (text &optional ignore-skim)
+   ;; Modified by Milan Zamazal <pdm@brailcom.org>:
+   ;; 2003-05-15: dtk-speaker-process handling removed
+   ;; 2003-05-27: text chunking removed
    "Speak the TEXT string on the  tts.
  This is achieved by sending the text to the speech server.
  No-op if variable `dtk-quiet' is set to nil.
***************
*** 565,574 ****
                      dtk-split-caps 
                      emacspeak-pronounce-pronunciation-table
                      selective-display ))
-                                         ; ensure  the process  is live
-   (unless (or (eq 'run (process-status dtk-speaker-process ))
-               (eq 'open (process-status dtk-speaker-process )))
-     (dtk-initialize))
                                          ; If you dont want me to talk,
                                          ;or my server is not
                                          ;running, I will remain silent.
--- 568,573 ----
***************
*** 616,642 ****
            (when pronunciation-table
              (emacspeak-pronounce-apply-pronunciations pronunciation-table))
            (dtk-quote mode))
!         (goto-char (point-min))
!         (skip-syntax-forward inherit-chunk-separator-syntax)
!         (while (and (not (eobp))
!                     (dtk-move-across-a-chunk
!                      inherit-chunk-separator-syntax
!                      complement-separator))
!                                         ;if we matched a punctuation,
!                                         ;treat this as a chunk only if the punctuation is followed
!                                         ;by white space
! 					;dtk-speak-treat-embedded-punctuations-specially
! 					;has been T for a long time
!           (unless
!               (and (char-after  (point))
!                    (= (char-syntax (preceding-char )) ?.)
!                    (not (= 32 (char-syntax (following-char )))))
!             (setq end (point ))
!             (dtk-format-text-and-speak  start end )
!             (setq start  end)))         ; end while
!                                         ; process trailing text
!         (or  (= start (point-max))
!              (dtk-format-text-and-speak start (point-max)))))
      (dtk-force)))
  
  (defun dtk-speak-list (text )
--- 615,621 ----
            (when pronunciation-table
              (emacspeak-pronounce-apply-pronunciations pronunciation-table))
            (dtk-quote mode))
! 	(dtk-format-text-and-speak (point-min) (point-max))))
      (dtk-force)))
  
  (defun dtk-speak-list (text )
***************
*** 672,685 ****
                                          ;Say these words, ie speak each of them as a word.
                                          ; Text is sent via the appropriate TCLSH.
  (defun dtk-say (words)
    "Say these WORDS."
    (declare (special dtk-speaker-process dtk-stop-immediately
                      dtk-speak-server-initialized dtk-quiet))
    ;; I wont talk if you dont want me to
    (unless dtk-quiet
-     (or (eq 'run (process-status dtk-speaker-process ))
-         (eq 'open (process-status dtk-speaker-process ))
-         (dtk-initialize))
      (when dtk-speak-server-initialized
        (and dtk-stop-immediately (dtk-stop ))
        (dtk-interp-say words))))
--- 651,663 ----
                                          ;Say these words, ie speak each of them as a word.
                                          ; Text is sent via the appropriate TCLSH.
  (defun dtk-say (words)
+   ;;; Modified on 2003-05-15 by Milan Zamazal <pdm@brailcom.org>, to allow
+   ;;; use with speechd-el -- dtk-speaker-process handling removed.
    "Say these WORDS."
    (declare (special dtk-speaker-process dtk-stop-immediately
                      dtk-speak-server-initialized dtk-quiet))
    ;; I wont talk if you dont want me to
    (unless dtk-quiet
      (when dtk-speak-server-initialized
        (and dtk-stop-immediately (dtk-stop ))
        (dtk-interp-say words))))
